File: src/sequence/swapMinAndMax/SwapMinAndMax.java
package sequence.swapMinAndMax;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;
import java.util.stream.Collectors;

public class SwapMinAndMax {
    public static void swapMinAndMax(String input) {
        List<Integer> numbers = Arrays.stream(input.split("\\s+"))
                .map(Integer::parseInt)
                .collect(Collectors.toList());
        Integer mx = Collections.max(numbers);
        Integer mn = Collections.min(numbers);
        List<String> swapedNumbers = numbers.stream()
                .map(n -> {
                    if (n == mx) {
                        n = mn;
                    } else if (n == mn) {
                        n = mx;
                    }
                    return String.valueOf(n);
                })
                .collect(Collectors.toList());
        System.out.println(String.join(" ", swapedNumbers));
    }

}

File: src/sequence/swapMinAndMax/swap_min_and_max.py
def swap_min_and_max(string):
    numbers = string.split()
    mn = min(numbers, key=int)
    mx = max(numbers, key=int)
    for i in range(len(numbers)):
        if numbers[i] == mn:
          numbers[i] = mx
        elif numbers[i] == mx:
          numbers[i] = mn
    print(*numbers)
